common/src/main/java/org/valkyrienskies/mod/mixin/entity/MixinEntity.java

```diff
@@ -153,6 +207,24 @@ private void preCalculateViewVector(final float xRot, final float yRot, final Ca
         cir.setReturnValue(newViewVector);
     }
 
+    @Inject(method = "shouldRender", at = @At("HEAD"), cancellable = true)
+    private void onShouldRender(double d, double e, double f, CallbackInfoReturnable<Boolean> cir) {
+        if (this.draggingInformation.isEntityBeingDraggedByAShip() && this.level.isClientSide) {
```

Comment:
add comment: why do we need to use the dragging information here? why doesn't it automatically work having replaced the position of the player with the correct position?

Reply:
Explained via comment.